node/ChainwebNode.hs

```diff
@@ -161,6 +166,47 @@ runCutMonitor logger db = L.withLoggerLabel ("component", "cut-monitor") logger
             -- $ S.map _cutMap
             -- $ cutStream db
 
+
+runAmberdataBlockMonitor :: Logger logger => logger -> CutDb cas -> IO ()
+runAmberdataBlockMonitor logger db = L.withLoggerLabel ("component", "amberdata-block-monitor") logger $ \l -> do
+    go l `catchAllSynchronous` \e ->
+        logFunctionText l Error ("Amberdata Block Monitor failed: " <> sshow e)
+    logFunctionText l Info "Stopped Amberdata Block Monitor"
+  where
+    go l = do
+      logFunctionText l Info "Initialized Amberdata Block Monitor"
+      void
+        $ S.mapM_ (logAllBlocks l)
+        $ S.map cutToAmberdataBlocks
+        $ cutStream db
+
+    logAllBlocks :: Logger logger => logger -> [AmberdataBlock] -> IO ()
+    logAllBlocks l = mapM_ (logFunctionJson l Info)
+
+    cutToAmberdataBlocks :: Cut -> [AmberdataBlock]
+    cutToAmberdataBlocks c = map
+                               (\(_,bh) -> AmberdataBlock
+                                           (uniqueBlockHeight bh totalChains)
+                                           (_blockHash bh)
+                                           (_blockCreationTime bh)
+                                           (_blockParent bh)
+                                           (_blockNonce bh)
+                                           (_blockMiner bh)
+                                           -- undefined Blocksize
+                                           -- undefined blockTransNum
+                                           (_blockChainId bh)
+                                           (_blockWeight bh)
+                               )
+                               (HM.toList (_cutMap c))
```

Comment:
This approach may miss some blocks, while reporting other bocks more than once. Cuts don't increase strictly in chains, they can also skip over blocks, and are not even monotonic for individual chains.

We should write a function that takes a stream of new cuts and emits a stream all new blocks. Such a function would use the last observed cut and the current cut and compute the set of new blocks. It is easy to write using `Chainweb.TreeDB.getBranch`.

Are we fine with missing some blocks? Otherwise are we planning to address this for this PR or for a follow up PR?

Reply:
I think I'd rather get this PR in now.  Missing blocks isn't ideal, but doing it in a follow-up PR is fine.

Reply:
I take care of this in a followup.